examples/fluids/advection2d.h

```diff
@@ -381,7 +410,116 @@ CEED_QFUNCTION(IFunction_Advection2d)(void *ctx, CeedInt Q,
+//                (1 - (P_star/P)^((gamma-1)/2 gamma)) / (gamma-1)
+//
+//    The above values are applied weakly on the inflow boundary faces.
+//
+// *****************************************************************************
+CEED_QFUNCTION(Advection2d_Sur)(void *ctx, CeedInt Q,
+                            const CeedScalar *const *in,
+                            CeedScalar *const *out) {
+  // *INDENT-OFF*
+  // Inputs
+  const CeedScalar (*q)[CEED_Q_VLA] = (const CeedScalar(*)[CEED_Q_VLA])in[0],
+                   (*qdataSur)[CEED_Q_VLA] = (const CeedScalar(*)[CEED_Q_VLA])in[1];
+  // Outputs
+  CeedScalar (*v)[CEED_Q_VLA] = (CeedScalar(*)[CEED_Q_VLA])out[0];
+  // *INDENT-ON*
+  SurfaceContext context = (SurfaceContext)ctx;
+  const CeedScalar cv = context->cv;
+  const CeedScalar cp = context->cp;
+  const CeedScalar Rd = context->Rd;
+  const CeedScalar P_wind = context->P_wind;
+  const CeedScalar rho_wind = context->rho_wind;
+  const CeedScalar strong_form = context->strong_form;
+  const CeedScalar *wind = context->wind;
+  const CeedScalar gamma = cp / cv;
+  const PetscBool implicit = context->implicit;
 
+  CeedPragmaSIMD
+  // Quadrature Point Loop
+  for (CeedInt i=0; i<Q; i++) {
+    // Setup
+    // -- Interp in
+    const CeedScalar rho        =    q[0][i];
+    const CeedScalar u[3]       =   {q[1][i] / rho,
+                                     q[2][i] / rho,
+                                     q[3][i] / rho
+                                    };
+    const CeedScalar E          =    q[4][i];
+    // -- Interp-to-Interp qdata
+    const CeedScalar wdetJb     =    qdataSur[0][i];
+    // ---- Normal vectors
+    const CeedScalar norm[2]    =   {qdataSur[1][i],
+                                     qdataSur[2][i]
+                                    };
+    // windDotNorm = dot(velocity, normal)
+    const CeedScalar windDotNorm = norm[0]*wind[0] + norm[1]*wind[1];
+    // Normal velocity
+    const CeedScalar u_n = norm[0]*u[0] + norm[1]*u[1];
+    // ke = kinetic energy inside the domain
+    const CeedScalar ke  = (u[0]*u[0] + u[1]*u[1]) / 2.;
+    // -- Values on inflow boundaries
+    // P_star = P, so the incoming nonlinear wave has amplitude zero
+    const CeedScalar P_star   = (E - ke * rho) * (gamma - 1.);
+    const CeedScalar u_star   = u_n + 2.*sqrt(fabs(gamma*P_star/rho))/(gamma - 1.);
+    const CeedScalar rho_star = rho_wind * pow(fabs(P_star/P_wind), (1./gamma));
+    const CeedScalar T_star   = P_star/(Rd*rho_star);
+    const CeedScalar E_star   = rho_star * (cv*T_star + u_star*u_star/2.);
+
+    // No Change in density or momentum
+    for (CeedInt j=0; j<4; j++) {
+      v[j][i] = 0;
+    }
+    // Implementing in/outflow BCs
+    switch (implicit) {
+    case 0: {  //explicit
+      if (windDotNorm >  1E-5) { // outflow
+        v[4][i] = -(1-strong_form) *wdetJb *E *u_n;
+      } else if (windDotNorm < -1E-5) { // inflow
+        v[4][i] = (1-strong_form) *wdetJb *E_star *u_star;
```

Comment:
I would expect the inflow `E_star` to be whatever inflow value is prescribed. With this, we get a wave coming in from the left.
```suggestion
        v[4][i] = (1-strong_form) *wdetJb * 1.0 *u_star;
```
I ran with
```
 ./navierstokes -problem advection2d -problem_advection_wind translation -problem_advection_wind_translation 1,0,0 -ts_max_time 50 -dm_plex_box_faces 32,32 -ts_adapt_monitor
```

Reply:
Thanks, @jedbrown . Yep, It is working, even in 3D.
I am very surprised though. This formulation used to work before this latest change in the setup. Could it be for the fact that we are not implementing slip/wall BCs anymore?!

Well, since it was working previously, I never doubted the correctness or compatibility of the inflow formulations with our problem. But, now I see that the value of `E_star` is negative because the initial `E` is Zero. I have updated `index.rst`, if that is easier to read. Could it be fundamentally wrong?

Currently, `P_wind` and `rho_wind` are prescribed by the user and `E_star` is computed accordingly. Do you think I should go ahead and prescribe `E_star` instead or do you have other suggestions?